src/middlewared/middlewared/plugins/virt/volume.py

```diff
@@ -70,6 +71,14 @@ async def do_create(self, data):
                 f'Not a valid storage pool. Choices are: {", ".join(global_config["storage_pools"])}'
             )
 
+        ds_name = os.path.join(target_pool, f'.ix-virt/custom/default_{data["name"]}')
+        if await self.middleware.call(
+            'zfs.dataset.query', [['id', '=', ds_name]], {
+                'extra': {'retrieve_children': False, 'retrieve_properties': False}
+            }
+        ):
+            verrors.add('virt_volume_create.name', 'ZFS dataset against this volume name already exists')
```

Comment:
I wonder if we should trigger a `virt.global.recover` in this case to rebuild the incus database for the incus storage pool. This will cause the stray zvol to get added in properly.

Reply:
You can see an example here: https://github.com/truenas/middleware/pull/16011